pkg/crypto/crypto.go

```diff
@@ -827,8 +827,12 @@ func (ca *CA) MakeClientCertificate(certFile, keyFile string, u user.Info, expir
 }
 
 func (ca *CA) MakeClientCertificateForDuration(u user.Info, lifetime time.Duration) (*TLSCertificateConfig, error) {
+	return ca.MakeClientCertificateForSubjectAndDuration(userToSubject(u), lifetime)
+}
+
+func (ca *CA) MakeClientCertificateForSubjectAndDuration(subject pkix.Name, lifetime time.Duration) (*TLSCertificateConfig, error) {
```

Comment:
what are you setting that cannot be a user.Info?  `userToSubject` has some warnings about golang issues with certain orderings.

Reply:
The installer is currently setting `OrganizationalUnit` (and no groups): https://github.com/openshift/installer/blob/master/pkg/asset/tls/etcd.go#L25

Reply:
Your public method should check to see that this subject conforms to rules here: https://github.com/openshift/library-go/blob/9723292f0698a984407307fa241880ea8b5ca9a0/pkg/crypto/crypto.go#L867-L879